### PR TITLE
Adds Puppeteer Tests for Google, Proton, and Skiff

### PR DESCRIPTION
**Pull request by jonathansampson**
This is a fairly large PR, so I won't be heart-broken if there are several items which require further attention 🙂 This PR delivers initial support for Puppeteer-Driven tests on Google Calendar, Proton Calendar, and Skiff Calendar.

Addresses #33

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5924,11 +7154,18 @@
       "integrity": "sha512-CC1bOL87PIWSBhDcTrdeLo6eGT7mCFtrg0uIJtqJUFyK+eJnzl8A1niH56uu7KMa5XFrtiV+AQuHO3n7DsHnLQ==",
       "dev": true
     },
+    "node_modules/word-wrap": {
+      "version": "1.2.3",
```
[the remark this answers is not in the document]
Comment (jonathansampson) [reply package-lock.json:7158]: Having an issue where `npm audit` claims that `npm audit fix` will resolve the issue, but `npm audit fix` fails to update the packages. I assume this behavior is related to https://github.com/npm/cli/issues/3472.

[the remark this answers is not in the document]
Comment (tackley) [reply package-lock.json:7158]: @jonathansampson you can address by forcing a later version of optionator that doesn't rely on the "vulnerable" version of word-wrap. This was the PR on talk to address: https://github.com/brave/brave-talk/pull/1073

**Review by tackley — COMMENTED** [review]
Other than a few minor nits to address, this looks great!

File: package.json [hunk 2]
```diff
@@ -4,28 +4,31 @@
...
-    "test": "jest",
+    "test": "jest --verbose --runInBand",
     "build": "webpack --mode production",
     "watch": "webpack --mode development --devtool inline-source-map --watch",
     "format": "prettier --write src/",
     "format:check": "prettier --check src/",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "create-env": "tsc ./src/scripts/createEnv.ts && npm run format && node ./src/scripts/createEnv.js",
+    "postinstall": "npm run create-env"
```
[the remark this answers is not in the document]
Comment (jonathansampson) [reply package.json:14]: Good idea. I was thinking about this yesterday when encountering some friction introduced by using `postinstall`. I'll instead remove the auto-create bits, and drop-in an .env.example.

**Review by tackley — APPROVED** [review]

**Pull request merged**